contracts/Governance/GovernorBravoDelegateG2.sol

```diff
@@ -91,24 +92,24 @@ contract GovernorBravoDelegate is GovernorBravoDelegateStorageV2, GovernorBravoE
         uint endBlock = add256(startBlock, votingPeriod);
 
         proposalCount++;
-        Proposal memory newProposal = Proposal({
-            id: proposalCount,
-            proposer: msg.sender,
-            eta: 0,
-            targets: targets,
-            values: values,
-            signatures: signatures,
-            calldatas: calldatas,
-            startBlock: startBlock,
-            endBlock: endBlock,
-            forVotes: 0,
-            againstVotes: 0,
-            abstainVotes: 0,
-            canceled: false,
-            executed: false
-        });
-
-        proposals[newProposal.id] = newProposal;
+        Proposal storage newProposal = proposals[proposalCount];
```

Comment:
This is incorrect since it doesn't create a new proprosal, but only creates link to a previous one.
You can test [this](https://github.com/grGred/Anyswap-staking/blob/main/contracts/test/testPointer.sol)

Reply:
Your test is using `memory` structs while this is using `storage`. This is effectively creating a pointer to the struct at the array index in memory and modifying the fields there.

Reply:
Okay, nvm

Reply:
.